### PR TITLE
Feature/15 indexing exclusions

### PR DESCRIPTION
**Pull request by blabadi**
implements #16

### CODE REVIEW

File: maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/rules/StaticExclusionRulesDAO.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Map;
+
+/**
+ * static rules dao, good for testing
+ */
+public class StaticExclusionRulesDAO implements ExclusionRulesDAO {
+
+    @SneakyThrows
+    public Mono<Map<Class<?>, List<? extends ExclusionRule>>> getExclusionRules() {
+        val clzz = Sample.class;
```
Comment (andricDu) [inline maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/rules/StaticExclusionRulesDAO.java]: not a fan of `clazz`? :smile:

File: maestro-app/src/test/java/bio/overture/maestro/app/infra/adapter/inbound/ManagementControllerTest.java [hunk 2]
```diff
@@ -168,21 +160,41 @@ void indexStudy() throws InterruptedException {
         assertEquals(expectedDoc0, docs.get(0));
     }
 
-
     @Test
+
```
Comment (andricDu) [inline maestro-app/src/test/java/bio/overture/maestro/app/infra/adapter/inbound/ManagementControllerTest.java]: is this extra line supposed to be here?

File: maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/rules/ApplicationPropertiesExclusionRulesDAO.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    public Mono<Map<Class<?>, List<? extends ExclusionRule>>> getExclusionRules() {
+        return Mono.just(this.exclusionRules);
+    }
+
+    @PostConstruct
+    public void init() throws Exception {
+        val mapper = new ObjectMapper(new YAMLFactory());
+        try {
+            this.exclusionRulesResource.getFile();
+        } catch (FileNotFoundException __) {
```
Comment (andricDu) [inline maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/rules/ApplicationPropertiesExclusionRulesDAO.java]: :+1:

File: maestro-app/src/test/java/bio/overture/maestro/app/infra/adapter/inbound/ManagementControllerTest.java (hunk 2 above)
Comment (blabadi) [reply maestro-app/src/test/java/bio/overture/maestro/app/infra/adapter/inbound/ManagementControllerTest.java]: nope will remove it

**Pull request merged**